### ISSUE TITLE
Get Pyston working on OSX

### ISSUE THREAD

**Issue by kmod** [pyston/pyston#165]
There are a number of issues, such as toolchain and I'm not sure what else.  Trying to create a single task to track them.

**Comment by dagar** [comment]
Working on a cmake configure script in PR #436.

**Comment by jzoppi** [comment]
So afaik there is space for porting to Win32. Is llvm ported to win32?
Best Regards,
Giorgio


2015-04-04 3:34 GMT+02:00 Daniel Agar <notifications@github.com>:

> Working on a cmake configure script in PR #436
> <https://github.com/dropbox/pyston/pull/436>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dropbox/pyston/issues/165#issuecomment-89479566>.
>



-- 
Quiero ser el rayo de sol que cada día te despierta
para hacerte respirar y vivir en me.
"Favola -Moda".

**Comment by kmod** [comment]
I would definitely recommend holding off on attempting a windows port until we get a mac version working: I think the windows porting effort will be much higher and we haven't even discovered all the issues on the Mac side.  We also haven't thought much about 32-bit support; if you'd like to help with porting, 32-bit linux might be a better place to start and would be very appreciated :)

**Comment by undingen** [comment]
Just want to mention: (I saw it on the weekend when I tried to fix the
object cache issue) I think llvm currently only supports patchpoints for
x86 in 64bit mode. So one would have to patch llvm too in order to get good
performance... but I'm not sure whats missing
Am 06.04.2015 17:53 schrieb "Kevin Modzelewski" <notifications@github.com>:

**Comment by dagar** [comment]
It sounds like libunwind is going to be a problem. See #456

From the libunwind-devel mailing list January 2015. http://lists.nongnu.org/archive/html/libunwind-devel/2015-01/msg00006.html

    "There's no OS X support in libunwind at the moment, and the platform is quite different."

**Comment by mhsjlw** [comment]
It seems apple has their own libunwind? http://opensource.apple.com/source/libunwind/

**Comment by kmod** [comment]
I think when we tried using it in the past, it didn't support a feature that Pyston currently uses.  That particular feature we use (remote unwinding) came as a surprise to us though, and I think we should be able to move off of it.  Or maybe Apple has implemented the feature since we last tried it.  Anyway, that's where we left off.

I don't think we've looked into OSX support for about two years so the situation should be much different now -- in particular dagar finished his work on the cmake system so that should be able to ease a bunch of the cross-platform issues.

It would be a huge help to us if anyone wanted to take another crack at this :)
